api/src/Middleware/CiabMiddleware.php

```diff
@@ -25,7 +38,21 @@ public function __invoke(Request $request, Response $response, $next)
             \loadAccount($user);
             $_SESSION['accountId'] = $user;
         }
-        $response = $next($request, $response);
+        try {
+            $response = $next($request, $response);
+        } catch (\App\Controller\NotFoundException $e) {
```

Comment:
Would it be possible to move these exceptions to another, separate file, and maybe let the `BaseController` extend from those? Then, in the middleware we could just reference the separated exception types rather than tying our middleware specifically to our controller.

So for example, it might look like:

`Exceptions.php`
- definition for NotFound
- definition for PermissionDenied
- etc

`BaseController.php`
- ControllerNotFound extends NotFound
- ControllerPermissionDenied extends PermissionDenied
- etc

`Middleware.php`, whichever route we take with in here vs separate file
- checks against `Exceptions\NotFound` rather than the `BaseController` implementation
- checks against `Exceptions\PermissionDenied` rather than the `BaseController` implementation
- etc

It would also help for the case where our other layers (Service, Repository) need to throw exceptions. As we move more code to that pattern, they won't want to inherit from or use `BaseController` exceptions. This comes with the benefit of allowing us the flexibility to update our pattern over time without breaking the ability of existing code from being able to `throw` on errors.